bindings/pyroot/cppyy/CPyCppyy/src/TemplateProxy.cxx

```diff
@@ -142,6 +142,12 @@ PyObject* TemplateProxy::Instantiate(const std::string& fname,
             } else
                 PyErr_Clear();
 
+            if (!bArgSet && (Py_TYPE(itemi) == &TemplateProxy_Type)) {
+                TemplateProxy *tp = (TemplateProxy*)itemi;
+                PyObject *tmpl_name = CPyCppyy_PyText_FromFormat("decltype(%s%s)", tp->fTI->fCppName.c_str(), tp->fTemplateArgs ? CPyCppyy_PyText_AsString(tp->fTemplateArgs) : "");
```

Comment:
I realize I do not understand that. Are we parsing some strings to pipe to `decltype`?

Reply:
Yes. Because of the way type lookups work in cling, we need that `decltype`. In InterOp the `decltype` is not required. That is; here the lookup happens with a parse step, and that requires `decltype`, whereas in InterOp we can lookup the name/type directly and do not require the `decltype`.

Reply:
Can we avoid somehow the that parsing. Don't we have the Decl around?

Reply:
Ok, sorry, I am reviewing the wrong fork. Ignore my comment ;) We should fix that on the compiler-research end though...